### PR TITLE
implement scanner

### PR DESCRIPTION
**Pull request by alex337**
implement scanner

### CODE REVIEW

File: pkg/ptree/scanner.go [hunk 1]
```diff
@@ -1,5 +1,132 @@
...
+	pod       *Pod
+	container *Container
+	process   *Process
+	node      *Node
+}
+
+type CgroupName []string
+
+func (scan *ScannerImpl) Scan(UID, QOS string) (*Pod, error){
+	return scan.getContainers(NewPod(UID, QOS)), nil
```
Comment (payall4u) [inline pkg/ptree/scanner.go:39]: Should return the right error here

File: pkg/ptree/scanner.go [hunk 2]
```diff
@@ -1,5 +1,132 @@
...
 }
+
+func (scan *ScannerImpl)readContainerFile(podPath string, pod *Pod) map[string]*Container {
+	fileList, err := ioutil.ReadDir(podPath)
+	if err != nil {
+		klog.Errorf("can't read %s, %v", podPath, err)
+		return nil
+	}
+	for _,file :=range fileList {
+		if len(file.Name()) == 64 {
```
Comment (payall4u) [inline pkg/ptree/scanner.go:95]: You can use regular expressions here.

File: pkg/ptree/scanner.go [hunk 3]
```diff
@@ -1,5 +1,132 @@
...
+			if err, processes :=scan.readPidFile(procPath, *scan.pod.Containers[containerId]); err != nil{
+				scan.pod.Containers[containerId].Processes = processes
+			}
+		}
+	}
+	return scan.pod.Containers
+
+}
+
+func (scan *ScannerImpl)readPidFile(procPath string, container Container) (error, map[int]*Process) {
```
Comment (payall4u) [inline pkg/ptree/scanner.go:112]: return (map, error)

File: pkg/ptree/scanner.go [hunk 4]
```diff
@@ -1,5 +1,132 @@
...
+
+}
+
+func (scan *ScannerImpl) getContainers(p *Pod) *Pod {
+	podPath := scan.getPodPath(p.UID, p.QOS)
+	var containers map[string]*Container
+	containers = make(map[string]*Container)
+
+	baseDir := filepath.Clean(filepath.Join(CgroupBase, podPath))
+	filepath.Walk(baseDir, func(path string, info os.FileInfo, err error) error {
```
Comment (payall4u) [inline pkg/ptree/scanner.go:49]: Needn't use walk here. 🛩️

File: pkg/ptree/scanner.go (hunk 1 above)
Comment (alex337) [reply pkg/ptree/scanner.go:39]: ok

File: pkg/ptree/scanner.go (hunk 2 above)
Comment (alex337) [reply pkg/ptree/scanner.go:95]: ok

File: pkg/ptree/scanner.go (hunk 3 above)
Comment (alex337) [reply pkg/ptree/scanner.go:112]: ok

File: pkg/ptree/scanner.go (hunk 4 above)
Comment (alex337) [reply pkg/ptree/scanner.go:49]: ok, i get it

**Pull request merged**